woocommerce/woocommerce-gutenberg-products-block
# Graduate `__experimental_woocommerce_blocks_checkout_update_order_meta` action to stable

This PR will deprecate the `__experimental_woocommerce_blocks_checkout_update_order_meta` action and replace it with `woocommerce_blocks_checkout_update_order_meta`. 

It also updates the docs relating to this hook. All references to the now deprecated hook have been _replaced_ with the new hook, besides the one in the experimental interfaces document, which has changed to note that the prefixed hook is now deprecated.


Fixes #4995 

### Manual Testing

How to test the changes in this Pull Request:

1. Add the following code somewhere that it'll be executed, for example your theme's `functions.php` file:
```php
add_action( '__experimental_woocommerce_blocks_checkout_update_order_meta', function( $order ) { return; }, 10, 1 );

// For testing purposes, update the order date using the new hook.
add_action( 'woocommerce_blocks_checkout_update_order_meta', function( $order ) {
	$order->set_date_created( '2000-05-08' );
	$order->save();
} );
```
2. Add the following directives to your `wp-config.php` file:
```php
define( 'WP_DEBUG', true );
define( 'WP_DEBUG_LOG', true );
```
3. Add a product to your cart and check out using the Checkout block. Ensure it goes through correctly.
4. Check your `debug.log` file and ensure there's a deprecation message logged for `__experimental_woocommerce_blocks_checkout_update_order_meta`.
5. Ensure the order creation date is May 8th 2000.



### Changelog

> Deprecate the __experimental_woocommerce_blocks_checkout_update_order_meta action in favour of woocommerce_blocks_checkout_update_order_meta.
